src/utransport.rs

```diff
@@ -38,103 +38,45 @@ pub trait LocalUriProvider: Send + Sync {
 ///
-/// Implementations of `UListener` contain the details for what should occur when a message is received
+/// Implementations contain the details for what should occur when a message is received.
 ///
-/// For more information, please refer to [uProtocol Specification](https://github.com/eclipse-uprotocol/up-spec/blob/main/up-l1/README.adoc).
-///
-/// # Examples
-///
-/// ## Simple example
-///
-/// ```
-/// use up_rust::{UListener, UMessage, UStatus};
-///
-/// use async_trait::async_trait;
-/// use std::sync::{Arc, Mutex};
-///
-/// #[derive(Clone)]
-/// struct FooListener {
-///     inner_foo: Arc<Mutex<String>>
-/// }
-///
-/// #[async_trait]
-/// impl UListener for FooListener {
-///     async fn on_receive(&self, msg: UMessage) {
-///         let mut inner_foo = self.inner_foo.lock().unwrap();
-///         if let Some(payload) = msg.payload.as_ref() {
-///             *inner_foo = format!("latest message length: {}", payload.len());
-///         }
-///     }
-/// }
-/// ```
-///
-/// ## Long-running function needed when message received
-///
-/// ```
-/// use up_rust::{UListener, UMessage, UStatus};
-///
-/// use async_trait::async_trait;
-/// use tokio::task;
-/// use std::sync::{Arc, Mutex};
-///
-/// #[derive(Clone)]
-/// struct LongTaskListener;
-///
-/// async fn send_to_jupiter(message_for_jupiter: UMessage) {
-///     // send a message to Jupiter
-///     println!("Fly me to the moon... {message_for_jupiter}");
-/// }
-///
-/// #[async_trait]
-/// impl UListener for LongTaskListener {
-///     async fn on_receive(&self, msg: UMessage) {
-///         tokio::spawn(send_to_jupiter(msg));
-///     }
-/// }
-/// ```
+/// Please refer to the [uProtocol Transport Layer specification](https://github.com/eclipse-uprotocol/up-spec/blob/v1.6.0-alpha.2/up-l1/README.adoc)
+/// for details.
 #[cfg_attr(test, automock)]
 #[async_trait]
 pub trait UListener: Send + Sync {
     /// Performs some action on receipt of a message.
     ///
     /// # Parameters
     ///
-    /// * `msg` - The message
+    /// * `msg` - The message to process.
     ///
-    /// # Note for `UListener` implementers
+    /// # Implementation hints
     ///
-    /// `on_receive()` is expected to return almost immediately. If it does not, it could potentially
-    /// block further message receipt. For long-running operations consider passing off received
-    /// data to a different async function to handle it and returning.
-    ///
-    /// # Note for `UTransport` implementers
-    ///
-    /// Because `on_receive()` is async you may choose to either `.await` it in the current context
-    /// or spawn it onto a new task and await there to allow current context to immediately continue.
+    /// This function is expected to return almost immediately. If it does not, it could potentially
+    /// block processing of succeeding messages. Long-running operations for processing a message should
+    /// therefore be run on a separate thread.
```

Comment:
'... thread' - or rather 'as a separate task'?

Reply:
If you want to run blocking code, then you should probably use a dedicated (and separate) thread for running the code. Whether that code is scheduled as a task or run by means of message passing, is not relevant in this context FMPOV.

Reply:
The thought was that 'task' represents a bigger option space for that might be done - essentially the same argument that you're making, but my word associations between task and thread seem to be the other way around :-)